lib/holidays/core_extensions/date.rb

```diff
@@ -29,6 +29,24 @@ def holiday?(*options)
         holidays && !holidays.empty?
       end
 
+      # Returns a new Date where one or more of the elements have been changed according to the +options+ parameter.
+      # The +options+ parameter is a hash with a combination of these keys: <tt>:year</tt>, <tt>:month</tt>, <tt>:day</tt>.
+      #
+      #   Date.new(2007, 5, 12).change(day: 1)               # => Date.new(2007, 5, 1)
+      #   Date.new(2007, 5, 12).change(year: 2005, month: 1) # => Date.new(2005, 1, 12)
+      def change(options)
```

Comment:
😍 

Oh my goodness. The PR is worth it *just* for this set of additions. Very very nice!